pkg/service/configcache/service.go

```diff
@@ -80,7 +81,18 @@ func (svc *Service) Read(clusterID uuid.UUID, host string) (NodeConfig, error) {
 		return emptyConfig, err
 	}
 
-	rawHostConfig, ok := clusterConfig.Load(host)
+	hostKey := host
+	// Remove leading '[' and trailing ']' if present.
+	// netip.ParseAddr(bracketedIP) leaves brackets.
+	// Need to remove it.
+	if host != "" && host[0] == '[' && host[len(host)-1] == ']' {
+		host = host[1 : len(host)-1]
+	}
+	if addr, err := netip.ParseAddr(host); err == nil {
+		hostKey = addr.String()
+	}
```

Comment:
So we still need to be cautious about different IP `string` representations and we do some parsing manually.
Wouldn't it be safer to change the map key type from `string` to `netip.Addr` which is also comparable?

Reply:
The host IP is not really configurable. It's possible to set agent's port in scylla-manager-agent.yaml, but not possible to set it per host in scylla-manager.yam.